orttraining/orttraining/training_ops/cpu/optimizer/gradient_control.cc

```diff
@@ -81,5 +81,63 @@ ONNX_OPERATOR_KERNEL_EX(
         .TypeConstraint("T2", DataTypeImpl::AllTensorTypes()),
     ZeroGradient<float>);
 
+ONNX_OPERATOR_KERNEL_EX(
+    InPlaceAccumulatorV2,
+    kMSDomain,
+    1,
+    kCpuExecutionProvider,
+    KernelDefBuilder()
+        .Alias(0, 1)  // accumulate tensors in-place
+        .TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
+    InPlaceAccumulatorV2<float>);
+
+template <typename T>
+Status InPlaceAccumulatorV2<T>::Compute(OpKernelContext* context) const {
+  Tensor* accumulation_buffer = const_cast<Tensor*>(context->Input<Tensor>(0));
+  const Tensor* new_value = context->Input<Tensor>(1);
+  const Tensor* overwrite_tensor = context->Input<Tensor>(2);
+
+  void* accumulation_buffer_data = accumulation_buffer->template MutableData<T>();
+  const bool overwrite = overwrite_tensor != nullptr ? *(overwrite_tensor->template Data<bool>()) : false;
+
+  if (overwrite) {
+    const void* updated_data = new_value->template Data<T>();
+    memcpy(accumulation_buffer_data, updated_data, new_value->SizeInBytes());
+  } else {
+    // Copy from Add CPU kernel
+    ProcessBroadcastSpanFuncs funcs{
```

Comment:
curious if it is possible we reuse same ProcessBroadcastSpanFuncs instance for both v1 and v2 kernels.

Reply:
Maybe but for that I can think of one way by holding a static ProcessBroadcastSpanFuncs  which will be used by both, but not sure if it is worth it. let me know if any other better way